[Bugfix] Apply RMSNorm weight correction for Gemma2 GGUF models

kitaekatt opened the pull request:
## Summary

llama.cpp adds 1 to RMSNorm weights during GGUF conversion (see [convert_hf_to_gguf.py#L3397-L3400](https://github.com/ggml-org/llama.cpp/blob/be7c3034108473beda214fd1d7c98fd6a7a3bdf5/convert_hf_to_gguf.py#L3397-L3400)), but vLLM expects original values. Without this correction, Gemma2 GGUF models produce gibberish output.

This fix applies the same correction that was added for Gemma3 in PR #26189 - subtracting 1 from norm weights during GGUF loading.

## Root Cause

When loading Gemma2 GGUF models, the RMSNorm weights are +1 higher than expected because llama.cpp's GGUF conversion adds 1 to these weights. The Gemma3 model already handles this (added in #26189), but Gemma2 was missing the same correction.

## Changes

- Add `_process_weights` generator to `Gemma2ForCausalLM.load_weights()` that subtracts 1 from norm weights when loading GGUF models
- Pattern matches existing Gemma3 implementation

## Testing

Tested with `bartowski/gemma-2-2b-it-GGUF` on RTX 5090:

**Before fix**: Gibberish output (excessive script mixing: ARABIC, CJK, CYRILLIC, GREEK, LATIN)
**After fix**: Coherent output, 40% accuracy on MMLU benchmark, 344 tok/s throughput

```
Single-Thread Baseline Results: 40.00% accuracy (4/10 correct)
Throughput: 343.70 tok/s
```

kitaekatt committed:
fix(gguf): Apply RMSNorm weight correction for Gemma2 GGUF models

llama.cpp adds 1 to RMSNorm weights during GGUF conversion (see
convert_hf_to_gguf.py#L3397-L3400), but vLLM expects original values.
Without this correction, Gemma2 GGUF models produce gibberish output.

This fix applies the same correction that was added for Gemma3 in
PR #26189 - subtracting 1 from norm weights during GGUF loading.

Fixes gibberish output from bartowski/gemma-2-2b-it-GGUF and similar
Gemma2 GGUF models on Blackwell GPUs (RTX 5090).

Signed-off-by: Christina <kitaekatt@gmail.com>
Signed-off-by: Christina <truffle@gmail.com>

hmellor reviewed (changes requested):
Instead of modifying the weights can we just say that `if self.quant_config.get_name() == "gguf"` we use `RMSNorm` instead of `GemmaRMSNorm`?

kitaekatt committed:
fix(gguf): Use RMSNorm instead of GemmaRMSNorm for GGUF Gemma2/3

GGUF files store RMSNorm weights with +1 already baked in (llama.cpp
convention). GemmaRMSNorm adds 1 in its forward pass, causing double
addition for GGUF models.

Instead of subtracting 1 during weight loading, select the norm class
at construction time: use plain RMSNorm (no +1) for GGUF, GemmaRMSNorm
for non-GGUF. Applied to all norm layers in Gemma2 and Gemma3 including
decoder layer norms and Gemma3's q_norm/k_norm in attention.

Removes the weight correction workaround from load_weights().

Addresses reviewer feedback from hmellor on PR #31464.

Signed-off-by: Christina <truffle@gmail.com>
